### PR TITLE
Adding PropTypes

### PR DESCRIPTION
**Pull request by HyrumG**

### CODE REVIEW

**Comment by joeycozza** [comment]
did this have any adverse affects when running storybook? I'm pretty sure some of the stories don't provide all of the props.

File: packages/react-scripts/template/src/components/example/FrontierDocsCard.js [hunk 1]
```diff
@@ -32,4 +33,13 @@ const FrontierDocsCard = ({ logoColor, animationDuration, handleLogoClick }) =>
   </Card>
 )
 
+FrontierDocsCard.propTypes = {
+  /** The logoColor is used to set the color of the Logo component. */
+  logoColor: PropTypes.string.isRequired,
```
Comment (joeycozza) [inline packages/react-scripts/template/src/components/example/FrontierDocsCard.js]: this isn't required. it defaults to black I think.
also, the animationDuration also have a default.

**Comment by HyrumG** [comment]
> did this have any adverse affects when running storybook? I'm pretty sure some of the stories don't provide all of the props.

`

File: packages/react-scripts/template/src/components/example/FrontierDocsCard.js (hunk 1 above)
Comment (gmdayley) [reply packages/react-scripts/template/src/components/example/FrontierDocsCard.js]: agreed, not required

File: packages/react-scripts/template/src/components/example/Logo.js [hunk 2]
```diff
@@ -42,4 +43,11 @@ const Logo = ({ color = 'black', animationDuration = 20 }) => {
   )
 }
 
+Logo.propTypes = {
+  /** Used to set the color of the logo. */
+  color: PropTypes.string.isRequired,
+  /** Used to set the duration of animation of the logo. */
+  animationDuration: PropTypes.number.isRequired,
```
Comment (gmdayley) [inline packages/react-scripts/template/src/components/example/Logo.js]: not required

File: packages/react-scripts/template/src/components/example/NotSignedInCard.js [hunk 3]
```diff
@@ -14,4 +15,11 @@ const NotSignedInCard = ({ user: { userLoading } }) => (
   </Card>
 )
 
+NotSignedInCard.propTypes = {
+  user: PropTypes.shape({
+    /** True or false depending on whether or not the user is being fetched. */
+    userLoading: PropTypes.bool.isRequired,
```
Comment (gmdayley) [inline packages/react-scripts/template/src/components/example/NotSignedInCard.js]: I dont believe this(`userLoading`) is required either

File: packages/react-scripts/template/src/components/example/Logo.js (hunk 2 above)
Comment (HyrumG) [reply packages/react-scripts/template/src/components/example/Logo.js]: `.isRequired` is not required or PropTypes for Logo is not required?

Comment (joeycozza) [reply packages/react-scripts/template/src/components/example/Logo.js]: the .isRequired

**Pull request merged**